src/class/hid/hid.h

```diff
@@ -222,7 +234,25 @@ typedef enum
   GAMEPAD_BUTTON_MODE   = TU_BIT(12), ///< Mode button
   GAMEPAD_BUTTON_THUMBL = TU_BIT(13), ///< L3 button
   GAMEPAD_BUTTON_THUMBR = TU_BIT(14), ///< R3 button
-//GAMEPAD_BUTTON_       = TU_BIT(15), ///< Undefined button
+  GAMEPAD_BUTTON_15     = TU_BIT(15), ///< Button 15
+#if (CFG_TUD_MAX_BUTTONS > 16)
+  GAMEPAD_BUTTON_17     = TU_BIT(16), ///< Button 17
```

Comment:
off by one, also incorrect Button 16 in comment for typo

Reply:
I have no problems making 32 buttons the only option. 

The reason I went the long way around was that I though people might be bothered by the increase in  size of the joystick report.

I also updated the GAMEPAD_BUTTON_ entries and comments. GAMEPAD_BUTTON_15 is now GHAMEPAD_BUTTON_16 to match how windows and most game reference the buttons.

Reply:
> I have no problems making 32 buttons the only option.
> 
> The reason I went the long way around was that I though people might be bothered by the increase in size of the joystick report.

Yeah, but don't worry, people shouldn't be affected if using the helper API. Those that uses the raw API should know what to do :)

> I also updated the GAMEPAD_BUTTON_ entries and comments. GAMEPAD_BUTTON_15 is now GHAMEPAD_BUTTON_16 to match how windows and most game reference the buttons.

I am not sure if this is a correct way to do it, @nekuneko what do you think. Should we start the button from 0 or from 1. I would love to hear your thought since you are the original author that add enhance support for gamepad.

Reply:
@hathach I considered starting it at 0, but since the other buttons have symbolic names (A, B, SELECT, START, etc) I think it makes more sense to match how they are displayed to the end user.

Reply:
thanks, that makes sense for sure. Though I still like to hear from another user on another platform (Linux) as well. Let just wait for his input for a day or two.

Reply:
No problem.

At least one app in Linux does 0-31 :) I don't know any actual games to test on the pi though. 

![image](https://user-images.githubusercontent.com/23555060/121233276-c9b82100-c892-11eb-9647-42f7b97eefb3.png)

Reply:
I just did a quick check, and Steam controller setting seems to start from 0, and is spot-on with Button South. Now I am more lean toward starting from 0.
![image](https://user-images.githubusercontent.com/249515/121289270-ded29580-c90e-11eb-8ebf-ea2529dd3122.png)

Reply:
Should I go ahead and change them to be 0 based, or do you feel like we should wait a bit more? I really don't have a strong opinion on this one and would be ok with either.

Reply:
> Should I go ahead and change them to be 0 based, or do you feel like we should wait a bit more? I really don't have a strong opinion on this one and would be ok with either.

Yes, please do, also please remove the "Note: button number from 1 on host OS" as well. I think we will just merge it starting with Button0.

Reply:
Done.